core/templates/services/math-interactions.service.spec.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit test for MathInteractionsService
+ */
+
+import { TestBed } from '@angular/core/testing';
+
+import { MathInteractionsService } from 'services/math-interactions.service';
+
+describe('MathInteractionsService', () => {
+  let mathInteractionsService: MathInteractionsService = null;
+
+  beforeEach(() => {
+    mathInteractionsService = TestBed.get(MathInteractionsService);
+  });
+
+  it('should validate the answer correctly', function() {
```

Comment:
Add many more cases. This seems to poorly test the conditions (e.g. no tests for inequalities). Add more complicated example inputs. Also add cases with multiple violations (like equation with a trailing operator).

Reply:
Add more success cases too!

Reply:
Done!